pkgs/desktops/gnome-3/misc/gnome-screensaver/fix-dbus-service-dir.patch

```diff
@@ -0,0 +1,11 @@
+--- a/configure.ac  2019-01-20 20:28:01.309231507 +0100
++++ b/configure.ac  2019-01-20 20:31:54.927978927 +0100
+@@ -137,7 +137,7 @@
+ 
+ # Find out where the session service file goes
+ # The sad sed hack is recomended by section 27.10 of the automake manual.
+-DBUS_SESSION_SERVICE_DIR=`pkg-config --variable session_bus_services_dir dbus-1 | sed -e 's,/usr/share,${datarootdir},g'`
++DBUS_SESSION_SERVICE_DIR=${datarootdir}/dbus-1/services
```

Comment:
You can use `PKG_CONFIG_DBUS_1_SESSION_BUS_SERVICES_DIR = "${placeholder "out"}/share/dbus-1/services";` in the expression.

Reply:
I don't understand what that's for, I want to set the path directly instead of using the sed hack.

Reply:
The pkgconfig + sed hack is introduced by the Ubuntu patch that adds the dbus service. As you said it doesn't work, so I just replace it with the correct path directly. I don't see why changing what pkg-config returns would help.

Reply:
The correct way to solve this upstream would be `pkg-config --variable session_bus_services_dir dbus-1 --define-variable 'datarootdir=${datarootdir}'`.

Reply:
I don't understand why pkg-config needs to be involved at all. This is the original patch:

https://git.launchpad.net/ubuntu/+source/gnome-screensaver/tree/debian/patches/05_dbus_service.patch?h=ubuntu/cosmic#n12

Reply:
Well, `pkg-config` reads the value from `$(nix-build -A dbus.dev)/lib/pkgconfig/dbus-1.pc`, which resolves to `/etc/dbus-1/services`. We can override the returned value using the environment variable.

The correct way to solve this upstream would be `pkg-config --variable session_bus_services_dir dbus-1 --define-variable 'datadir=${datadir}'`, see https://www.bassi.io/articles/2018/03/15/pkg-config-and-paths/

Reply:
On Ubuntu if they want to get the variable from dbus's pkg-config they're doing the right thing because it has `session_bus_services_dir=/usr/share/dbus-1/services`. For us I don't think there's any point using pkg-config because `services.dbus.packages` only picks up services from certain documented paths, this being one of them.

Reply:
We use `pkg-config` so that we can be sure we are using the correct location. Sure, we can hardcode the path but then we will need to keep it synchronized.

Reply:
Even if we override the pkg-config variable we would still need to put in /usr/share because of the sed hack that's already ethere. Isn't it just cleaner to get rid of the hack, to set the right thing directly and forget pkg-config?

Reply:
Well, the sed will not bother us with the environment variable set:

```ShellSession
$ env PKG_CONFIG_DBUS_1_SESSION_BUS_SERVICES_DIR='${placeholder "out"}/share/dbus-1/services' pkg-config --variable session_bus_services_dir dbus-1 | sed -e 's,/usr/share,${datarootdir},g'
${placeholder "out"}/share/dbus-1/services
```

Reply:
OK, in that case isn't it best to patch in the correct solution you described? I've pushed that now.

Reply:
I guess we might want to switch to Arch’s patchset, as it is much closer to our philosophy. With Ubuntu, it is hard to tell which patches are Ubuntu specific, only needed due to patches of some other component (24 looks like that).

Reply:
24 is also in the Arch patches: https://git.archlinux.org/svntogit/community.git/tree/trunk/use-screensaver-background.patch?h=packages/gnome-screensaver

The enabled Ubuntu patches all seem to be bugfixes not present in the other distros, I think it would be a good idea to have them. I think if we want to really get it right we should support xscreensaver since this package is old and deprecated, but let's do that in another PR.